src/pathfinder.rs

```diff
@@ -0,0 +1,367 @@
+    }
+}
+
+impl std::fmt::Display for DeleteNodeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DeleteNodeError::MutexError(e) => write!(f, "Mutex error : {e}"),
+            DeleteNodeError::ParsingError(e) => write!(f, "Parsing error : {e}"),
+            DeleteNodeError::NodeNotFound => write!(f, "Node not found"),
+        }
+    }
+}
+
+byond_fn!(fn remove_node_astar(unique_id) {
+    let mut nodes = NODES.write().unwrap();
+    match remove_node(unique_id, nodes.as_mut()) {
+        Ok(s) => Some(s),
+        Err(e) => Some(format!("{e}"))
+    }
+});
+
+// Replace the node with unique_id by None
+// Update connected nodes as well so nothing target the removed node anymore
+// Errors if no node can be found with unique_id
+fn remove_node(unique_id: &str, nodes: &mut [NodeContainer]) -> Result<String, DeleteNodeError> {
+    let unique_id = unique_id.parse::<usize>()?;
+
+    {
+        let node_to_delete_container = nodes.get(unique_id).cloned();
+
+        let node_to_delete_ref = match node_to_delete_container {
+            None => return Err(DeleteNodeError::NodeNotFound),
+            Some(node_container) => node_container.node,
+        };
+
+        let node_to_delete_ref = node_to_delete_ref.as_ref().read().unwrap();
+
+        let node_to_delete = match node_to_delete_ref.as_ref() {
+            None => return Err(DeleteNodeError::NodeNotFound),
+            Some(node) => node,
+        };
+
+        // Erase all links to the removed node
+        node_to_delete.connected_nodes_id.iter().for_each(|i| {
+            if let Some(node_container) = nodes.get_mut(*i) {
+                if let Some(node) = node_container.node.write().unwrap().as_mut() {
+                    node.connected_nodes_id
+                        .retain(|index| index != &node_to_delete.unique_id);
+                }
+            }
+        });
+    } // We need to drop everything before set the removed node to None. This is to ensure memory safety
+
+    nodes[unique_id] = NodeContainer::new_empty();
+
+    Ok("1".to_string())
+}
+
+enum AstarError {
+    MutexError(mut_static::Error),
+    StartNodeNotFound,
+    GoalNodeNotFound,
+    NoPathFound,
+}
+
+impl From<mut_static::Error> for AstarError {
+    fn from(error: mut_static::Error) -> Self {
+        AstarError::MutexError(error)
+    }
+}
+
+byond_fn!(fn generate_path_astar(start_node_id, goal_node_id) {
+    let nodes = NODES.read().unwrap();
+    if let (Ok(start_node_id), Ok(goal_node_id)) = (start_node_id.parse::<usize>(), goal_node_id.parse::<usize>()) {
+        match generate_path(start_node_id, goal_node_id, &nodes) {
+            Ok(vector) => Some(match serde_json::to_string(&vector) {
+                Ok(s) => s,
+                Err(_) => "Cannot serialize path".to_string(),
+            }),
+            Err(e) => Some(match e {
+                AstarError::MutexError(e) => format!("Mutex error : {}", e),
```

Comment:
These should all just be a normal display function (which thiserror makes very easy).